In [1]:
import pandas as pd

In [2]:
cars = pd.read_csv("https://raw.githubusercontent.com/juliandnl/redi_ss20/master/cars.csv")

In [3]:
cars.head()

Make   Model  Year                           Variant     Kms   Price  \
0  Volkswagen   Vento  2012                  2.5 Luxury 170cv   99950  360000   
1        Ford  Ranger  2012                2.3 Cd Xl Plus 4x2  140000  320000   
2  Volkswagen     Fox  2011                     1.6 Trendline  132000  209980   
3        Ford  Ranger  2017  3.2 Cd Xls Tdci 200cv Automática   13000  798000   
4  Volkswagen     Gol  2013                1.4 Power 83cv 3 p  107000  146000   

   Doors       Kind           Location  
0    4.0      Sedán            Córdoba  
1    2.0    Pick-Up         Entre Ríos  
2    5.0  Hatchback  Bs.as. G.b.a. Sur  
3    4.0    Pick-Up            Neuquén  
4    3.0  Hatchback            Córdoba

In [5]:
cars.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 9 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Make      10000 non-null  object 
 1   Model     10000 non-null  object 
 2   Year      10000 non-null  int64  
 3   Variant   10000 non-null  object 
 4   Kms       10000 non-null  int64  
 5   Price     10000 non-null  int64  
 6   Doors     10000 non-null  float64
 7   Kind      10000 non-null  object 
 8   Location  10000 non-null  object 
dtypes: float64(1), int64(3), object(5)
memory usage: 703.2+ KB


In [6]:
cars.shape

(10000, 9)

In [20]:
cars[['Make', 'Model', 'Year', 'Location']].head(5)

Make   Model  Year           Location
0  Volkswagen   Vento  2012            Córdoba
1        Ford  Ranger  2012         Entre Ríos
2  Volkswagen     Fox  2011  Bs.as. G.b.a. Sur
3        Ford  Ranger  2017            Neuquén
4  Volkswagen     Gol  2013            Córdoba

1. Logical Operators

In [23]:
cars['Year'] >= 2016

0       False
1       False
2       False
3        True
4       False
        ...  
9995     True
9996    False
9997    False
9998    False
9999    False
Name: Year, Length: 10000, dtype: bool

In [24]:
cars[cars['Year'] >= 2016]

Make      Model  Year                             Variant    Kms  \
3           Ford     Ranger  2017    3.2 Cd Xls Tdci 200cv Automática  13000   
8           Ford  Focus III  2017                               1.6 S  17500   
29    Volkswagen    Saveiro  2017  1.6 Gp Ce 101cv Safety + Pack High  20000   
35          Ford  Focus III  2016                2.0 Sedan Se Plus Mt  15022   
49          Ford  Focus III  2016                               1.6 S  39000   
...          ...        ...   ...                                 ...    ...   
9985  Volkswagen  Gol Trend  2016                 1.6 Trendline 101cv  54100   
9990        Ford  Focus III  2016                      2.0 Se Plus Mt  48000   
9991        Ford     Ranger  2016        3.2 Cd 4x2 Xlt At Tdci 200cv  59000   
9994        Ford  Focus III  2016                              2.0 Se  18000   
9995        Ford  Focus III  2016                              2.0 Se  67000   

       Price  Doors       Kind               Location  
3     798000    4.0    Pick-Up                Neuquén  
8     367000    5.0  Hatchback  Buenos Aires Interior  
29    370000    2.0    Pick-Up                Córdoba  
35    550000    4.0      Sedán      Bs.as. G.b.a. Sur  
49    454800    5.0  Hatchback      Bs.as. G.b.a. Sur  
...      ...    ...        ...                    ...  
9985  270000    5.0  Hatchback  Buenos Aires Interior  
9990  450000    5.0  Hatchback      Bs.as. G.b.a. Sur  
9991  790000    4.0    Pick-Up    Bs.as. G.b.a. Oeste  
9994  370000    5.0  Hatchback        Capital Federal  
9995  399000    5.0  Hatchback    Bs.as. G.b.a. Oeste  

[1414 rows x 9 columns]

In [26]:
cars['Make'] == 'Ford'

0       False
1        True
2       False
3        True
4       False
        ...  
9995     True
9996    False
9997     True
9998     True
9999     True
Name: Make, Length: 10000, dtype: bool

In [27]:
cars[cars['Make'] == 'Ford']

Make                  Model  Year                           Variant  \
1     Ford                 Ranger  2012                2.3 Cd Xl Plus 4x2   
3     Ford                 Ranger  2017  3.2 Cd Xls Tdci 200cv Automática   
6     Ford                     Ka  2012                1.0 Fly Viral 63cv   
7     Ford                     Ka  2012                1.6 Fly Viral 95cv   
8     Ford              Focus III  2017                             1.6 S   
...    ...                    ...   ...                               ...   
9994  Ford              Focus III  2016                            2.0 Se   
9995  Ford              Focus III  2016                            2.0 Se   
9997  Ford  Fiesta Kinetic Design  2012         1.6 Design 120cv Titanium   
9998  Ford  Fiesta Kinetic Design  2013         1.6 Design 120cv Titanium   
9999  Ford              Focus III  2014                  2.0 Titanium At6   

         Kms   Price  Doors       Kind               Location  
1     140000  320000    2.0    Pick-Up             Entre Ríos  
3      13000  798000    4.0    Pick-Up                Neuquén  
6      95243  142000    3.0  Hatchback      Bs.as. G.b.a. Sur  
7     110000  148000    3.0  Hatchback  Buenos Aires Interior  
8      17500  367000    5.0  Hatchback  Buenos Aires Interior  
...      ...     ...    ...        ...                    ...  
9994   18000  370000    5.0  Hatchback        Capital Federal  
9995   67000  399000    5.0  Hatchback    Bs.as. G.b.a. Oeste  
9997   89000  250000    5.0  Hatchback                Tucumán  
9998   76000  295000    5.0  Hatchback  Buenos Aires Interior  
9999   32000  380000    5.0  Hatchback      Bs.as. G.b.a. Sur  

[4025 rows x 9 columns]

In [29]:
cars[(cars['Make'] == 'Ford') & (cars['Year'] >= 2016)]

Make      Model  Year                           Variant    Kms    Price  \
3     Ford     Ranger  2017  3.2 Cd Xls Tdci 200cv Automática  13000   798000   
8     Ford  Focus III  2017                             1.6 S  17500   367000   
35    Ford  Focus III  2016              2.0 Sedan Se Plus Mt  15022   550000   
49    Ford  Focus III  2016                             1.6 S  39000   454800   
57    Ford     Mondeo  2018          2.0 Se Ecoboost At 240cv  15000  1200000   
...    ...        ...   ...                               ...    ...      ...   
9949  Ford  Focus III  2017                             1.6 S  10000   530000   
9990  Ford  Focus III  2016                    2.0 Se Plus Mt  48000   450000   
9991  Ford     Ranger  2016      3.2 Cd 4x2 Xlt At Tdci 200cv  59000   790000   
9994  Ford  Focus III  2016                            2.0 Se  18000   370000   
9995  Ford  Focus III  2016                            2.0 Se  67000   399000   

      Doors       Kind               Location  
3       4.0    Pick-Up                Neuquén  
8       5.0  Hatchback  Buenos Aires Interior  
35      4.0      Sedán      Bs.as. G.b.a. Sur  
49      5.0  Hatchback      Bs.as. G.b.a. Sur  
57      4.0      Sedán             Entre Ríos  
...     ...        ...                    ...  
9949    5.0  Hatchback    Bs.as. G.b.a. Oeste  
9990    5.0  Hatchback      Bs.as. G.b.a. Sur  
9991    4.0    Pick-Up    Bs.as. G.b.a. Oeste  
9994    5.0  Hatchback        Capital Federal  
9995    5.0  Hatchback    Bs.as. G.b.a. Oeste  

[631 rows x 9 columns]

In [30]:
cars[(cars['Make'] == 'Volkswagen') | (cars['Year'] >= 2015)]

Make      Model  Year                            Variant     Kms  \
0     Volkswagen      Vento  2012                   2.5 Luxury 170cv   99950   
2     Volkswagen        Fox  2011                      1.6 Trendline  132000   
3           Ford     Ranger  2017   3.2 Cd Xls Tdci 200cv Automática   13000   
4     Volkswagen        Gol  2013                 1.4 Power 83cv 3 p  107000   
5     Volkswagen     Amarok  2014  2.0 Cd Tdi 180cv 4x4 Highline C34  115000   
...          ...        ...   ...                                ...     ...   
9992  Volkswagen      Suran  2011          1.6 Imotion Trendline 11b   80000   
9993        Ford   Ecosport  2015             2.0 Titanium 143cv 4x2   50000   
9994        Ford  Focus III  2016                             2.0 Se   18000   
9995        Ford  Focus III  2016                             2.0 Se   67000   
9996  Volkswagen       Bora  2012              1.9 Trendline I 100cv  120000   

       Price  Doors         Kind               Location  
0     360000    4.0        Sedán                Córdoba  
2     209980    5.0    Hatchback      Bs.as. G.b.a. Sur  
3     798000    4.0      Pick-Up                Neuquén  
4     146000    3.0    Hatchback                Córdoba  
5     790000    4.0      Pick-Up  Buenos Aires Interior  
...      ...    ...          ...                    ...  
9992  190000    5.0  Monovolumen    Bs.as. G.b.a. Norte  
9993  490000    5.0          SUV                Córdoba  
9994  370000    5.0    Hatchback        Capital Federal  
9995  399000    5.0    Hatchback    Bs.as. G.b.a. Oeste  
9996  240000    4.0        Sedán  Buenos Aires Interior  

[6526 rows x 9 columns]

2. query()

In [33]:
cars.query('Make == "Ford"')

Make                  Model  Year                           Variant  \
1     Ford                 Ranger  2012                2.3 Cd Xl Plus 4x2   
3     Ford                 Ranger  2017  3.2 Cd Xls Tdci 200cv Automática   
6     Ford                     Ka  2012                1.0 Fly Viral 63cv   
7     Ford                     Ka  2012                1.6 Fly Viral 95cv   
8     Ford              Focus III  2017                             1.6 S   
...    ...                    ...   ...                               ...   
9994  Ford              Focus III  2016                            2.0 Se   
9995  Ford              Focus III  2016                            2.0 Se   
9997  Ford  Fiesta Kinetic Design  2012         1.6 Design 120cv Titanium   
9998  Ford  Fiesta Kinetic Design  2013         1.6 Design 120cv Titanium   
9999  Ford              Focus III  2014                  2.0 Titanium At6   

         Kms   Price  Doors       Kind               Location  
1     140000  320000    2.0    Pick-Up             Entre Ríos  
3      13000  798000    4.0    Pick-Up                Neuquén  
6      95243  142000    3.0  Hatchback      Bs.as. G.b.a. Sur  
7     110000  148000    3.0  Hatchback  Buenos Aires Interior  
8      17500  367000    5.0  Hatchback  Buenos Aires Interior  
...      ...     ...    ...        ...                    ...  
9994   18000  370000    5.0  Hatchback        Capital Federal  
9995   67000  399000    5.0  Hatchback    Bs.as. G.b.a. Oeste  
9997   89000  250000    5.0  Hatchback                Tucumán  
9998   76000  295000    5.0  Hatchback  Buenos Aires Interior  
9999   32000  380000    5.0  Hatchback      Bs.as. G.b.a. Sur  

[4025 rows x 9 columns]

In [35]:
cars.query('Make == "Ford" & Year >= 2016')

Make      Model  Year                           Variant    Kms    Price  \
3     Ford     Ranger  2017  3.2 Cd Xls Tdci 200cv Automática  13000   798000   
8     Ford  Focus III  2017                             1.6 S  17500   367000   
35    Ford  Focus III  2016              2.0 Sedan Se Plus Mt  15022   550000   
49    Ford  Focus III  2016                             1.6 S  39000   454800   
57    Ford     Mondeo  2018          2.0 Se Ecoboost At 240cv  15000  1200000   
...    ...        ...   ...                               ...    ...      ...   
9949  Ford  Focus III  2017                             1.6 S  10000   530000   
9990  Ford  Focus III  2016                    2.0 Se Plus Mt  48000   450000   
9991  Ford     Ranger  2016      3.2 Cd 4x2 Xlt At Tdci 200cv  59000   790000   
9994  Ford  Focus III  2016                            2.0 Se  18000   370000   
9995  Ford  Focus III  2016                            2.0 Se  67000   399000   

      Doors       Kind               Location  
3       4.0    Pick-Up                Neuquén  
8       5.0  Hatchback  Buenos Aires Interior  
35      4.0      Sedán      Bs.as. G.b.a. Sur  
49      5.0  Hatchback      Bs.as. G.b.a. Sur  
57      4.0      Sedán             Entre Ríos  
...     ...        ...                    ...  
9949    5.0  Hatchback    Bs.as. G.b.a. Oeste  
9990    5.0  Hatchback      Bs.as. G.b.a. Sur  
9991    4.0    Pick-Up    Bs.as. G.b.a. Oeste  
9994    5.0  Hatchback        Capital Federal  
9995    5.0  Hatchback    Bs.as. G.b.a. Oeste  

[631 rows x 9 columns]

3. Indexing with .loc()

In [36]:
cars.loc[cars['Year'] >= 2016]

Make      Model  Year                             Variant    Kms  \
3           Ford     Ranger  2017    3.2 Cd Xls Tdci 200cv Automática  13000   
8           Ford  Focus III  2017                               1.6 S  17500   
29    Volkswagen    Saveiro  2017  1.6 Gp Ce 101cv Safety + Pack High  20000   
35          Ford  Focus III  2016                2.0 Sedan Se Plus Mt  15022   
49          Ford  Focus III  2016                               1.6 S  39000   
...          ...        ...   ...                                 ...    ...   
9985  Volkswagen  Gol Trend  2016                 1.6 Trendline 101cv  54100   
9990        Ford  Focus III  2016                      2.0 Se Plus Mt  48000   
9991        Ford     Ranger  2016        3.2 Cd 4x2 Xlt At Tdci 200cv  59000   
9994        Ford  Focus III  2016                              2.0 Se  18000   
9995        Ford  Focus III  2016                              2.0 Se  67000   

       Price  Doors       Kind               Location  
3     798000    4.0    Pick-Up                Neuquén  
8     367000    5.0  Hatchback  Buenos Aires Interior  
29    370000    2.0    Pick-Up                Córdoba  
35    550000    4.0      Sedán      Bs.as. G.b.a. Sur  
49    454800    5.0  Hatchback      Bs.as. G.b.a. Sur  
...      ...    ...        ...                    ...  
9985  270000    5.0  Hatchback  Buenos Aires Interior  
9990  450000    5.0  Hatchback      Bs.as. G.b.a. Sur  
9991  790000    4.0    Pick-Up    Bs.as. G.b.a. Oeste  
9994  370000    5.0  Hatchback        Capital Federal  
9995  399000    5.0  Hatchback    Bs.as. G.b.a. Oeste  

[1414 rows x 9 columns]

In [37]:
cars.loc[(cars['Make'] == 'Ford') & (cars['Year'] >= 2016)]

Make      Model  Year                           Variant    Kms    Price  \
3     Ford     Ranger  2017  3.2 Cd Xls Tdci 200cv Automática  13000   798000   
8     Ford  Focus III  2017                             1.6 S  17500   367000   
35    Ford  Focus III  2016              2.0 Sedan Se Plus Mt  15022   550000   
49    Ford  Focus III  2016                             1.6 S  39000   454800   
57    Ford     Mondeo  2018          2.0 Se Ecoboost At 240cv  15000  1200000   
...    ...        ...   ...                               ...    ...      ...   
9949  Ford  Focus III  2017                             1.6 S  10000   530000   
9990  Ford  Focus III  2016                    2.0 Se Plus Mt  48000   450000   
9991  Ford     Ranger  2016      3.2 Cd 4x2 Xlt At Tdci 200cv  59000   790000   
9994  Ford  Focus III  2016                            2.0 Se  18000   370000   
9995  Ford  Focus III  2016                            2.0 Se  67000   399000   

      Doors       Kind               Location  
3       4.0    Pick-Up                Neuquén  
8       5.0  Hatchback  Buenos Aires Interior  
35      4.0      Sedán      Bs.as. G.b.a. Sur  
49      5.0  Hatchback      Bs.as. G.b.a. Sur  
57      4.0      Sedán             Entre Ríos  
...     ...        ...                    ...  
9949    5.0  Hatchback    Bs.as. G.b.a. Oeste  
9990    5.0  Hatchback      Bs.as. G.b.a. Sur  
9991    4.0    Pick-Up    Bs.as. G.b.a. Oeste  
9994    5.0  Hatchback        Capital Federal  
9995    5.0  Hatchback    Bs.as. G.b.a. Oeste  

[631 rows x 9 columns]

In [42]:
cars.loc[(cars['Year'] >= 2018) & (cars['Year'] >= 2016)]

Make         Model  Year                          Variant  \
57             Ford        Mondeo  2018         2.0 Se Ecoboost At 240cv   
64       Volkswagen          Polo  2018                  1.6 Comfortline   
309      Volkswagen  Golf Variant  2018          1.4 Comfortline Tsi Dsg   
463   Mercedes Benz      Sprinter  2018  2.1 411 Street 116cv 3250 V1 Tn   
538      Volkswagen     Gol Trend  2018               1.6 Serie 101cv 5p   
...             ...           ...   ...                              ...   
9271           Ford     Focus III  2018                  2.0 Titanium Mt   
9533     Volkswagen        Amarok  2018   2.0 Cs Tdi 140cv Trendline 4x2   
9822           Ford        Ranger  2018            2.5 Cs Ivct Xlt 166cv   
9871           Ford     Focus III  2018                            1.6 S   
9963  Mercedes Benz      Sprinter  2018  2.1 411 Street 116cv 3250 V2 Tn   

        Kms    Price  Doors       Kind                Location  
57    15000  1200000    4.0      Sedán              Entre Ríos  
64     8733   470000    4.0      Sedán         Capital Federal  
309    8000   815000    5.0      Rural   Buenos Aires Interior  
463    7000   980000    2.0     Furgón  Bs.as. Costa Atlántica  
538   10000   250000    5.0  Hatchback       Bs.as. G.b.a. Sur  
...     ...      ...    ...        ...                     ...  
9271  12360   653400    5.0  Hatchback         Capital Federal  
9533  11700   710000    2.0    Pick-Up     Bs.as. G.b.a. Oeste  
9822   7000   730000    2.0    Pick-Up                Santa Fe  
9871  13000   605000    5.0  Hatchback     Bs.as. G.b.a. Oeste  
9963  11000   920000    2.0     Furgón     Bs.as. G.b.a. Oeste  

[98 rows x 9 columns]

In [43]:
cars.loc[(cars['Year'] >= 2018) | (cars['Year'] >= 2016)]

Make      Model  Year                             Variant    Kms  \
3           Ford     Ranger  2017    3.2 Cd Xls Tdci 200cv Automática  13000   
8           Ford  Focus III  2017                               1.6 S  17500   
29    Volkswagen    Saveiro  2017  1.6 Gp Ce 101cv Safety + Pack High  20000   
35          Ford  Focus III  2016                2.0 Sedan Se Plus Mt  15022   
49          Ford  Focus III  2016                               1.6 S  39000   
...          ...        ...   ...                                 ...    ...   
9985  Volkswagen  Gol Trend  2016                 1.6 Trendline 101cv  54100   
9990        Ford  Focus III  2016                      2.0 Se Plus Mt  48000   
9991        Ford     Ranger  2016        3.2 Cd 4x2 Xlt At Tdci 200cv  59000   
9994        Ford  Focus III  2016                              2.0 Se  18000   
9995        Ford  Focus III  2016                              2.0 Se  67000   

       Price  Doors       Kind               Location  
3     798000    4.0    Pick-Up                Neuquén  
8     367000    5.0  Hatchback  Buenos Aires Interior  
29    370000    2.0    Pick-Up                Córdoba  
35    550000    4.0      Sedán      Bs.as. G.b.a. Sur  
49    454800    5.0  Hatchback      Bs.as. G.b.a. Sur  
...      ...    ...        ...                    ...  
9985  270000    5.0  Hatchback  Buenos Aires Interior  
9990  450000    5.0  Hatchback      Bs.as. G.b.a. Sur  
9991  790000    4.0    Pick-Up    Bs.as. G.b.a. Oeste  
9994  370000    5.0  Hatchback        Capital Federal  
9995  399000    5.0  Hatchback    Bs.as. G.b.a. Oeste  

[1414 rows x 9 columns]


Conclusion:

There are 631 Ford cars produced in or after 2016 in our dataset.



2.Let's analyze cars based on price and kms by brands.

    How expensive are the car makes in general?
    Which car make is the cheapest one?
    Which car make has on average the most Kms?



In [44]:
# Mean price over all cars & brands:
cars.Price.mean()

349545.4545

In [45]:
# mean price per car brand
cars.groupby('Make')[['Price']].mean()

Price
Make                        
Chrysler       366845.833333
Ford           353041.877764
Honda          381520.603448
Mercedes Benz  512749.515625
Volkswagen     335441.534119

In [49]:
cars.groupby('Make')[['Price']].mean().round(1).sort_values('Price', ascending=False)

Price
Make                   
Mercedes Benz  512749.5
Honda          381520.6
Chrysler       366845.8
Ford           353041.9
Volkswagen     335441.5

We can also add another column to the groups. For example, what is the average kms by each make?

In [52]:
cars.groupby('Make')[['Price', 'Kms']].mean().round(1)

Price      Kms
Make                            
Chrysler       366845.8  79594.0
Ford           353041.9  69897.5
Honda          381520.6  81619.7
Mercedes Benz  512749.5  67358.7
Volkswagen     335441.5  77660.2

**Conclusiuon:**
- The most expensive car make is Mercedes Benz
- The cheapest car make is Volkswagen
- The most car used is Honda (the highest Km)

3.What are the average prices by brands and by the number of doors that cars have?

In [57]:
cars.groupby(['Make', 'Doors'])[['Price']].mean().round(1)

Price
Make          Doors          
Chrysler      5.0    366845.8
Ford          2.0    481503.9
              3.0    158703.6
              4.0    455984.9
              5.0    331894.6
Honda         3.0     26000.0
              4.0    323781.2
              5.0    431716.5
Mercedes Benz 2.0    574315.4
              4.0    497913.1
              5.0    476930.1
Volkswagen    2.0    379124.7
              3.0    226298.6
              4.0    422831.0
              5.0    271274.8

**Conclusion:**

- For Mercedes, the most expensive cars have 2 doors
- For Ford, the most expensive car have 2 doors


4.How many unique models does have each brand?


In [59]:
cars.groupby('Make')[['Model']].nunique()

Model
Make                
Chrysler           2
Ford              11
Honda              6
Mercedes Benz     15
Volkswagen        28

**Conclusion:**
- Volkswagenhave 28 diferent models
- Chrysler have only 2 models

5.Which kind of cars is the most expensive and most popular?

In [65]:
cars.groupby('Kind')[['Price']].agg(['mean', 'count']).round(1)

Price      
                 mean count
Kind                       
Cabriolet     79272.6    11
Coupé        543092.2    90
Furgón       702054.7    53
Hatchback    276554.8  4056
Minivan      650000.0     1
Monovolumen  270500.2   705
Pick-Up      545194.6  1547
Rural        330247.1    59
SUV          380708.0  1305
Sedán        338569.6  2173

In [66]:
cars.groupby('Kind')[['Price']].mean().round(1)

Price
Kind                 
Cabriolet     79272.6
Coupé        543092.2
Furgón       702054.7
Hatchback    276554.8
Minivan      650000.0
Monovolumen  270500.2
Pick-Up      545194.6
Rural        330247.1
SUV          380708.0
Sedán        338569.6


Conclusion

    The most popular kind is Hatchback and then Sedan.
    The most expensive kind is Furgon, while the cheapest one is Cabriolet.



BONUS: What are the min, max and average production year for Ford cars?

In [69]:
ford_cars = cars[cars['Make'] == 'Ford']
ford_cars['Year'].agg(['mean', 'min', 'max']).round(1)

mean    2013.5
min     2011.0
max     2018.0
Name: Year, dtype: float64

In [70]:
cars[cars['Make'] == 'Ford']['Year'].agg(['mean','min','max']).round(1)

mean    2013.5
min     2011.0
max     2018.0
Name: Year, dtype: float64


Sorting Data

Let's say, we would like to see the least used cars (in terms of kms).

In [73]:
cars.sort_values('Kms').head(5) # sort_values() sort ascending by default

Make      Model  Year                 Variant   Kms   Price  \
5665  Volkswagen    Saveiro  2017  1.6 Gp Cs 101cv Safety  5094  320000   
8673        Ford   Ecosport  2017         1.6 S 110cv 4x2  5100  390000   
8828  Volkswagen  Gol Trend  2017  1.6 Trendline 101cv 3p  5500  330000   
2461        Ford  Focus III  2017                  2.0 Se  5500  480000   
7910  Volkswagen    Saveiro  2017  1.6 Gp Cs 101cv Safety  5600  360000   

      Doors       Kind                Location  
5665    2.0    Pick-Up       Bs.as. G.b.a. Sur  
8673    5.0        SUV       Bs.as. G.b.a. Sur  
8828    3.0  Hatchback       Bs.as. G.b.a. Sur  
2461    5.0  Hatchback         Capital Federal  
7910    2.0    Pick-Up  Bs.as. Costa Atlántica

In [74]:
cars.sort_values('Kms', ascending = False).head(5)

Make   Model  Year                            Variant     Kms  \
5204        Ford  Ranger  2011                      3.0 Cd Xl 4x2  218000   
5793        Ford  Ranger  2011               3.0 Cd Superduty 4x4  215000   
7369        Ford  Ranger  2011                 3.0 Cd Xl Plus 4x2  215000   
4029  Volkswagen  Amarok  2011  2.0 Cd Tdi 4x2 Trendline Ll17 1t8  214000   
2049  Volkswagen  Amarok  2011   2.0 Cd Tdi 4x4 Highline Pack 1hp  209898   

       Price  Doors     Kind             Location  
5204  300000    2.0  Pick-Up    Bs.as. G.b.a. Sur  
5793  385000    2.0  Pick-Up    Bs.as. G.b.a. Sur  
7369  310000    2.0  Pick-Up              Córdoba  
4029  400000    4.0  Pick-Up               Chubut  
2049  550000    4.0  Pick-Up  Bs.as. G.b.a. Norte

If you want to overwrite the object with the new values ordered there are 2 ways to do it:

    1. Assign to cars the value cars.sort_values()



In [76]:
sorted_cars = cars.sort_values('Kms')
sorted_cars.head(5)

Make      Model  Year                 Variant   Kms   Price  \
5665  Volkswagen    Saveiro  2017  1.6 Gp Cs 101cv Safety  5094  320000   
8673        Ford   Ecosport  2017         1.6 S 110cv 4x2  5100  390000   
8828  Volkswagen  Gol Trend  2017  1.6 Trendline 101cv 3p  5500  330000   
2461        Ford  Focus III  2017                  2.0 Se  5500  480000   
7910  Volkswagen    Saveiro  2017  1.6 Gp Cs 101cv Safety  5600  360000   

      Doors       Kind                Location  
5665    2.0    Pick-Up       Bs.as. G.b.a. Sur  
8673    5.0        SUV       Bs.as. G.b.a. Sur  
8828    3.0  Hatchback       Bs.as. G.b.a. Sur  
2461    5.0  Hatchback         Capital Federal  
7910    2.0    Pick-Up  Bs.as. Costa Atlántica

2. Use the inplace parameter of .sort_values()

In [77]:
cars.sort_values('Kms', ascending=False, inplace=True)
cars.head(5)

Make   Model  Year                            Variant     Kms  \
5204        Ford  Ranger  2011                      3.0 Cd Xl 4x2  218000   
5793        Ford  Ranger  2011               3.0 Cd Superduty 4x4  215000   
7369        Ford  Ranger  2011                 3.0 Cd Xl Plus 4x2  215000   
4029  Volkswagen  Amarok  2011  2.0 Cd Tdi 4x2 Trendline Ll17 1t8  214000   
2049  Volkswagen  Amarok  2011   2.0 Cd Tdi 4x4 Highline Pack 1hp  209898   

       Price  Doors     Kind             Location  
5204  300000    2.0  Pick-Up    Bs.as. G.b.a. Sur  
5793  385000    2.0  Pick-Up    Bs.as. G.b.a. Sur  
7369  310000    2.0  Pick-Up              Córdoba  
4029  400000    4.0  Pick-Up               Chubut  
2049  550000    4.0  Pick-Up  Bs.as. G.b.a. Norte

Another way to sort data in Pandas is using sort_index(). As its name suggests this statement is used for sorting indexes.

In [78]:
cars.sort_index()

Make                  Model  Year  \
0     Volkswagen                  Vento  2012   
1           Ford                 Ranger  2012   
2     Volkswagen                    Fox  2011   
3           Ford                 Ranger  2017   
4     Volkswagen                    Gol  2013   
...          ...                    ...   ...   
9995        Ford              Focus III  2016   
9996  Volkswagen                   Bora  2012   
9997        Ford  Fiesta Kinetic Design  2012   
9998        Ford  Fiesta Kinetic Design  2013   
9999        Ford              Focus III  2014   

                               Variant     Kms   Price  Doors       Kind  \
0                     2.5 Luxury 170cv   99950  360000    4.0      Sedán   
1                   2.3 Cd Xl Plus 4x2  140000  320000    2.0    Pick-Up   
2                        1.6 Trendline  132000  209980    5.0  Hatchback   
3     3.2 Cd Xls Tdci 200cv Automática   13000  798000    4.0    Pick-Up   
4                   1.4 Power 83cv 3 p  107000  146000    3.0  Hatchback   
...                                ...     ...     ...    ...        ...   
9995                            2.0 Se   67000  399000    5.0  Hatchback   
9996             1.9 Trendline I 100cv  120000  240000    4.0      Sedán   
9997         1.6 Design 120cv Titanium   89000  250000    5.0  Hatchback   
9998         1.6 Design 120cv Titanium   76000  295000    5.0  Hatchback   
9999                  2.0 Titanium At6   32000  380000    5.0  Hatchback   

                   Location  
0                   Córdoba  
1                Entre Ríos  
2         Bs.as. G.b.a. Sur  
3                   Neuquén  
4                   Córdoba  
...                     ...  
9995    Bs.as. G.b.a. Oeste  
9996  Buenos Aires Interior  
9997                Tucumán  
9998  Buenos Aires Interior  
9999      Bs.as. G.b.a. Sur  

[10000 rows x 9 columns]

In [79]:
cars.sort_index(inplace = True)

In [80]:
cars

Make                  Model  Year  \
0     Volkswagen                  Vento  2012   
1           Ford                 Ranger  2012   
2     Volkswagen                    Fox  2011   
3           Ford                 Ranger  2017   
4     Volkswagen                    Gol  2013   
...          ...                    ...   ...   
9995        Ford              Focus III  2016   
9996  Volkswagen                   Bora  2012   
9997        Ford  Fiesta Kinetic Design  2012   
9998        Ford  Fiesta Kinetic Design  2013   
9999        Ford              Focus III  2014   

                               Variant     Kms   Price  Doors       Kind  \
0                     2.5 Luxury 170cv   99950  360000    4.0      Sedán   
1                   2.3 Cd Xl Plus 4x2  140000  320000    2.0    Pick-Up   
2                        1.6 Trendline  132000  209980    5.0  Hatchback   
3     3.2 Cd Xls Tdci 200cv Automática   13000  798000    4.0    Pick-Up   
4                   1.4 Power 83cv 3 p  107000  146000    3.0  Hatchback   
...                                ...     ...     ...    ...        ...   
9995                            2.0 Se   67000  399000    5.0  Hatchback   
9996             1.9 Trendline I 100cv  120000  240000    4.0      Sedán   
9997         1.6 Design 120cv Titanium   89000  250000    5.0  Hatchback   
9998         1.6 Design 120cv Titanium   76000  295000    5.0  Hatchback   
9999                  2.0 Titanium At6   32000  380000    5.0  Hatchback   

                   Location  
0                   Córdoba  
1                Entre Ríos  
2         Bs.as. G.b.a. Sur  
3                   Neuquén  
4                   Córdoba  
...                     ...  
9995    Bs.as. G.b.a. Oeste  
9996  Buenos Aires Interior  
9997                Tucumán  
9998  Buenos Aires Interior  
9999      Bs.as. G.b.a. Sur  

[10000 rows x 9 columns]


Exercise

Find which 10 cars has the highest prices. Please use sort_values() and head(). While finding the top ten expensive cars overwrite the dataset and then return back to original sorting by using sort_index().


In [81]:
cars.head()

Make   Model  Year                           Variant     Kms   Price  \
0  Volkswagen   Vento  2012                  2.5 Luxury 170cv   99950  360000   
1        Ford  Ranger  2012                2.3 Cd Xl Plus 4x2  140000  320000   
2  Volkswagen     Fox  2011                     1.6 Trendline  132000  209980   
3        Ford  Ranger  2017  3.2 Cd Xls Tdci 200cv Automática   13000  798000   
4  Volkswagen     Gol  2013                1.4 Power 83cv 3 p  107000  146000   

   Doors       Kind           Location  
0    4.0      Sedán            Córdoba  
1    2.0    Pick-Up         Entre Ríos  
2    5.0  Hatchback  Bs.as. G.b.a. Sur  
3    4.0    Pick-Up            Neuquén  
4    3.0  Hatchback            Córdoba

In [83]:
cars.sort_values('Price', ascending=False).head(10)

Make            Model  Year  \
8007  Mercedes Benz        Clase GLC  2017   
2632           Ford          Mustang  2018   
9520  Mercedes Benz          Clase E  2014   
5404  Mercedes Benz               ML  2013   
2032  Mercedes Benz          Clase E  2014   
7897     Volkswagen  Tiguan Allspace  2018   
5831  Mercedes Benz               ML  2011   
57             Ford           Mondeo  2018   
6434  Mercedes Benz          Clase A  2014   
9187  Mercedes Benz          Clase A  2014   

                                  Variant    Kms    Price  Doors       Kind  \
8007      2.0 Glc250 300 4matic Atomático  18000  2800000    5.0        SUV   
2632                         5.0 Gt 421cv   7000  2350000    2.0      Coupé   
9520   3.5 E350 Avantgarde Sport B.eff At  34000  1725000    4.0      Sedán   
5404  3.5 Ml350 4matic Sport B.efficiency  72000  1499990    5.0        SUV   
2032   3.5 E350 Avantgarde Sport B.eff At  59000  1390000    4.0      Sedán   
7897          1.4 Tsi Trendline 150cv Dsg  10000  1310000    5.0        SUV   
5831      3.5 Ml350 4matic Sport Facelift  31700  1200000    5.0        SUV   
57               2.0 Se Ecoboost At 240cv  15000  1200000    4.0      Sedán   
6434      2.0 A 250 At Sport B.efficiency  48255  1170000    5.0  Hatchback   
9187      2.0 A 250 At Sport B.efficiency  48455  1170000    5.0  Hatchback   

                 Location  
8007    Bs.as. G.b.a. Sur  
2632  Bs.as. G.b.a. Norte  
9520              Córdoba  
5404      Capital Federal  
2032      Capital Federal  
7897  Bs.as. G.b.a. Oeste  
5831              Neuquén  
57             Entre Ríos  
6434  Bs.as. G.b.a. Norte  
9187  Bs.as. G.b.a. Oeste

In [85]:
cars.sort_values('Price', ascending=False, inplace=True)
cars.head(10)

Make            Model  Year  \
8007  Mercedes Benz        Clase GLC  2017   
2632           Ford          Mustang  2018   
9520  Mercedes Benz          Clase E  2014   
5404  Mercedes Benz               ML  2013   
2032  Mercedes Benz          Clase E  2014   
7897     Volkswagen  Tiguan Allspace  2018   
5831  Mercedes Benz               ML  2011   
57             Ford           Mondeo  2018   
6434  Mercedes Benz          Clase A  2014   
9187  Mercedes Benz          Clase A  2014   

                                  Variant    Kms    Price  Doors       Kind  \
8007      2.0 Glc250 300 4matic Atomático  18000  2800000    5.0        SUV   
2632                         5.0 Gt 421cv   7000  2350000    2.0      Coupé   
9520   3.5 E350 Avantgarde Sport B.eff At  34000  1725000    4.0      Sedán   
5404  3.5 Ml350 4matic Sport B.efficiency  72000  1499990    5.0        SUV   
2032   3.5 E350 Avantgarde Sport B.eff At  59000  1390000    4.0      Sedán   
7897          1.4 Tsi Trendline 150cv Dsg  10000  1310000    5.0        SUV   
5831      3.5 Ml350 4matic Sport Facelift  31700  1200000    5.0        SUV   
57               2.0 Se Ecoboost At 240cv  15000  1200000    4.0      Sedán   
6434      2.0 A 250 At Sport B.efficiency  48255  1170000    5.0  Hatchback   
9187      2.0 A 250 At Sport B.efficiency  48455  1170000    5.0  Hatchback   

                 Location  
8007    Bs.as. G.b.a. Sur  
2632  Bs.as. G.b.a. Norte  
9520              Córdoba  
5404      Capital Federal  
2032      Capital Federal  
7897  Bs.as. G.b.a. Oeste  
5831              Neuquén  
57             Entre Ríos  
6434  Bs.as. G.b.a. Norte  
9187  Bs.as. G.b.a. Oeste